src/test/scala/uk/gov/hmrc/ui/specs/BaseSpec.scala

```diff
@@ -18,16 +18,23 @@ package uk.gov.hmrc.ui.specs
 
 import org.scalatest.featurespec.AnyFeatureSpec
 import org.scalatest.matchers.should.Matchers
-import org.scalatest.{BeforeAndAfterEach, GivenWhenThen}
-import uk.gov.hmrc.selenium.webdriver.{Browser, ScreenshotOnFailure}
+import org.scalatest.{AppendedClues, BeforeAndAfterEach, GivenWhenThen}
+import uk.gov.hmrc.selenium.webdriver.{Browser, Driver, ScreenshotOnFailure}
+import uk.gov.hmrc.ui.pages.BasePage
 
 trait BaseSpec
     extends AnyFeatureSpec
     with GivenWhenThen
     with Matchers
     with BeforeAndAfterEach
     with Browser
-    with ScreenshotOnFailure {
+    with ScreenshotOnFailure
+    with AppendedClues {
+
+  def userShouldSee[P <: BasePage](page: P): Unit = {
+    Driver.instance.getCurrentUrl shouldBe page.url withClue "open browser page should have expected url"
+    Driver.instance.getTitle      shouldBe page.expectedPageTitle withClue "open browser page should have expected page title"
+  }
```

Comment:
I think "page.getPageTitle()" which is essentially "page.getTitleOfBrowser()" is a bit indirect - page objects need a browser to do certain things, but not sure a page object should be responsible for telling us stuff about the browser we're using during the test, the browser is something provisioned by the test

it might be nice to have the checks in the page itself, rather than the test needing to know how to say if page is current one - just did it here for simplicity

guess alternatively you can have page objects throwing exceptions with the clues above as the messages to indicate what went wrong